### PR TITLE
Browser based debugger

### PR DESCRIPTION
**Pull request by richfitz**
I don't think anyone would use this from dust directly, but it will be nice from odin!

Not too much magic in the end, though some more tweaks will be required to export things like interpolators later.

The basic idea from the user perspective is outlined in the docs included here: the model runs and dumps them in `browser()` but via some mild sleight of hand we have copied over some quantities from the system they were running!  This will make the most sense when put behind some conditional statements (as the example here does).

This is almost (if not entirely) impossible to test automatically, because it really requires interactivity and mocking is not possible because the call through to `browser` comes from a call in C++.  The example included shows the basic idea though

### CODE REVIEW

File: R/browser.R [hunk 1]
```diff
@@ -0,0 +1,124 @@
+##' Control browser-based of dust models.  This help page documents
```
Comment (weshinsley) [inline R/browser.R:1]:
```suggestion
##' Control browser-based debugging of dust models.  This help page documents
```

File: R/browser.R [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+##' from presssing `<enter>`.
+##'
+##' * You can press `n` or `c` to proceed to the next enabled iteration
+##' * You can press `Q` to quit the browser (this will end up as an
+##'   error by the time you have control back)
+##'
+##' These commands are established by `browser`, and can't be
+##' disabled.  This means that if you have an variable called `n` you
+##' will need to work with it as `(n)` (i.e. in parentheses).  This
+##' applies to all of browser's command variables (`c`, `f`, `n`, `s`,
```
Comment (weshinsley) [inline R/browser.R:21]:
```suggestion
##' applies to all of `browser`'s command variables (`c`, `f`, `n`, `s`,
```

Comment (weshinsley) [reply R/browser.R:21]: although this might look clunky

File: R/browser.R [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+##' `r` and `Q`); please see [browser] for more information.
+##'
+##' By the time the environment has been created, some variables from
+##' your model will have been copied into the environment; you can see
+##' these by running `ls()` and write expressions involving these
+##' objects.  Changes that you make in R are *not* (currently)
+##' propagated back into the running system.
```
Comment (weshinsley) [inline R/browser.R:28]: Does the "currently" imply you're thinking of an option to copy the debug environment back to the running one? Interesting...

File: R/browser.R [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+##'
+##' If you enable the debugger, you may have very many iterations to
+##' get through before control is returned back to the console.  You
+##' can run `dust_debug_continue()` to prevent entry into the debugger
+##' until control is passed back to the you; this means the time
```
Comment (weshinsley) [inline R/browser.R:33]:
```suggestion
##' until control is passed back to you; this means the time
```

**Review by weshinsley — APPROVED** [review]
I like this one! A few doc fixes, but all good!

File: R/browser.R (hunk 3 above)
Comment (richfitz) [reply R/browser.R:28]: it does mean exactly that! not actually that hard to do

**Pull request merged**